nilearn/datasets/_utils.py

```diff
@@ -347,31 +347,31 @@ def uncompress_file(file_, delete_archive=True, verbose=1):
             z.extractall(path=data_dir)
             z.close()
             if delete_archive:
-                os.remove(file_)
-            file_ = filename
+                file_.unlink()
+            file_ = Path(filename)
             processed = True
         elif file_.suffix == ".gz" or header.startswith(b"\x1f\x8b"):
             import gzip
 
             if file_.suffix == ".tgz":
-                filename = Path(f"{filename}.tar")
+                filename = f"{filename}.tar"
             elif file_.suffix == "":
                 # We rely on the assumption that gzip files have an extension
                 shutil.move(file_, f"{file_}.gz")
-                file_ = f"{file_}.gz"
+                file_ = Path(f"{file_}.gz")
             with gzip.open(file_) as gz:
                 with open(filename, "wb") as out:
                     shutil.copyfileobj(gz, out, 8192)
             # If file is .tar.gz, this will be handled in the next case
             if delete_archive:
-                os.remove(file_)
-            file_ = filename
+                file_.unlink()
+            file_ = Path(filename)
```

Comment:
```suggestion
```
if things go well it should still be a Path

Reply:
```suggestion
            file_ = filename
```

Reply:
@Remi-Gau this line should stay, just no need to cast to Path.